xml_converter/generators/cpp_templates/class_template.cpp

```diff
@@ -149,55 +149,28 @@ waypoint::{{cpp_class}} {{cpp_class}}::as_protobuf() const {
-                    }
-                {% else: %}
-                    if (trigger.{{attribute_variable.protobuf_field}}() != 0) {
-                        this->{{attribute_variable.attribute_name}} = trigger.{{attribute_variable.protobuf_field}}();
-                        this->{{attribute_variable.attribute_flag_name}} = true;
-                    }
-                {% endif %}
-            {% else: %}
-                {% if (attribute_variable.attribute_type ==  "Enum") %}
-                    if (proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}() != 0) {
-                        this->{{attribute_variable.attribute_name}} = from_proto_{{attribute_variable.class_name}}(proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}());
-                        this->{{attribute_variable.attribute_flag_name}} = true;
-                    }
-                {% elif (attribute_variable.attribute_type in ["MultiflagValue", "CompoundValue", "Custom", "CompoundCustomClass"])%}
-                    if (proto_{{cpp_class_header}}.has_{{attribute_variable.protobuf_field}}()) {
-                        this->{{attribute_variable.attribute_name}} = from_proto_{{attribute_variable.class_name}}(proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}());
-                        this->{{attribute_variable.attribute_flag_name}} = true;
-                    }
-                {% elif (attribute_variable.class_name == "string") %}
-                    if (proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}() != "") {
-                        this->{{attribute_variable.attribute_name}} = proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}();
-                        this->{{attribute_variable.attribute_flag_name}} = true;
-                    }
-                {% else: %}
-                    if (proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}() != 0) {
-                        this->{{attribute_variable.attribute_name}} = proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}();
-                        this->{{attribute_variable.attribute_flag_name}} = true;
-                    }
-                {% endif %}
+            {% if (attribute_variable.attribute_type in ["MultiflagValue", "CompoundValue", "Custom", "CompoundCustomClass"]) %}
+                if (proto_{{cpp_class_header}}{{attribute_variable.proto_drilldown_calls}}.has_{{attribute_variable.protobuf_field}}()) {
+                    this->{{attribute_variable.attribute_name}} = from_proto_{{attribute_variable.class_name}}(proto_{{cpp_class_header}}{{attribute_variable.proto_drilldown_calls}}.{{attribute_variable.protobuf_field}}());
+                    this->{{attribute_variable.attribute_flag_name}} = true;
+                }
+            {% elif (attribute_variable.class_name == "string") %}{# TODO: why is this .class_name when the others are .attribute_name #}
```

Comment:
I think this TODO is meant to say `attribute_type` instead of name. I think it can be changed to `attribute_type` without issue since `class_name` is directly assigned by `attribute_type` in the generator for String types. Can be a separate PR

Reply:
Yes it is supposed to be `attribute_type`. It will changed in a later PR.